### ISSUE TITLE
False AS0024 when removing obsolete overloaded procedure

### ISSUE THREAD

**Issue by jwikman** [microsoft/AL#5871]
**Describe the bug**
If I remove an overloaded procedure that is marked as obsolete (either obsolete attribute on procedure, or ObsoleteState Pending on object) I get AS0024 error

**To Reproduce**
Steps and to reproduce the behavior:
Create a codeunit with a couple of procedures that are overloaded (same name, different signatures)
Set Obsolete attribute on one of the procedures
Compile to get an app file
Copy the app file to the .alpackages folder
Configure appsourcecop.json for that version
Increase version in app.json
Delete the obsoleted procedure or set it as internal and compile
Now AS0022 will appear. `error AS0024: The number of parameters in the external procedure 'MyOverloadedProcedure' has changed from '1' to '0'. This will break dependent extensions.` 

If I set the procedure to internal, instead of deleting, I also get AS0022, reported in #5870

v1.0
``` 
codeunit 50100 "asdf MyCodeunit"
{
    procedure MyOverloadedProcedure()
    begin
        Message('repro');
    end;

    [Obsolete('will be removed', '15.0')]
    procedure MyOverloadedProcedure(param: Integer)
    begin
        Message('now');
    end;
}
```
v1.1 -> AS0024
``` 
codeunit 50100 "asdf MyCodeunit"
{
    procedure MyOverloadedProcedure()
    begin
        Message('repro');
    end;
}
```
v1.1 with internal -> AS0024 + AS0022
``` 
codeunit 50100 "asdf MyCodeunit"
{
    procedure MyOverloadedProcedure()
    begin
        Message('repro');
    end;

    internal procedure MyOverloadedProcedure(param: Integer)
    begin
        Message('now');
    end;
}
```

**Expected behavior**
We should be able to delete or change the scope of a procedure if the procedure was obsoleted

**Screenshots**
N/A

**5. Versions:**

 - AL Language: 5.0.270354

 - Business Central: N/A

**Comment by qutreson** [comment]
@jwikman we just published a new version of the AL extension 5.0.271898, it should fix issues related to the obsoletion of overloaded methods, can you give it a try?

**Comment by jwikman** [comment]
@qutreson I can confirm that this one is fixed in 5.0.271898
Thanks!
